firebase-firestore/src/main/java/com/google/firebase/firestore/remote/FirebaseClientGrpcMetadataProvider.java

```diff
@@ -45,8 +45,12 @@ public DefaultGrpcMetadata(
   @Override
   public void updateMetadata(@NonNull Metadata metadata) {
     metadata.put(USER_AGENT_HEADER, userAgentPublisher.getUserAgent());
-    metadata.put(
-        HEART_BEAT_HEADER,
-        Integer.toString(heartBeatInfo.getHeartBeatCode(firebaseFirestoreHeartBeatTag).getCode()));
+    int heartBeatCode = heartBeatInfo.getHeartBeatCode(firebaseFirestoreHeartBeatTag).getCode();
+    if (heartBeatCode != 0) {
+      metadata.put(
+          HEART_BEAT_HEADER,
+          Integer.toString(
+              heartBeatInfo.getHeartBeatCode(firebaseFirestoreHeartBeatTag).getCode()));
```

Comment:
Re-use `heartBeatCode` here? Otherwise it's very likely that you'll end up sending zero every time here since the first call consumed the beat for the day.

Reply:
reused nice catch!